### PR TITLE
Fix sort indicator for DataTables

### PR DESCRIPTION
**Pull request by Wikiwix**
## Description

As per [material spec ascending order should be shown via an upward arrow](https://material.io/components/data-tables#anatomy:~:text=Sorting%20by%20the%20schedule%20column%20in%20ascending%20order%20is%20indicated%20with%20an%20upward%20arrow.)
This PR changes the displayed arrows accordingly. for DataTables as well as PaginatedDataTables

In theory adding a single `!` to inverse the `ascending` value would have had the same effect as my suggested change. My change tries to keep the code concise and intuitive.
I kept the original formatting to make the changes easier to review, although `dartfmt` would suggest quite some "whitespace" changes.

## Related Issues

* #62745 

## Tests

A test to check that the arrow indicator has the proper orientation was added.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Wikiwix** [comment]
@googlebot I signed it!

**Comment by Wikiwix** [comment]
@googlebot I signed it!

**Comment by Wikiwix** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fflutter%2Fflutter%2Fpull%2F62795) for more info**.

@googlebot I signed it!

**Comment by Hixie** [comment]
Change looks good but it does need a test. In theory you would think that this change should have broken one, but I guess our test coverage was insufficient in this area.

**Comment by Wikiwix** [comment]
@Hixie Ok, I will have a look, but it is going to be a while as this will be my first UI test… 🙈

**Comment by Wikiwix** [comment]
@Hixie Added a unit test as requested :)
Hope it is what you expected.

File: packages/flutter/test/material/data_table_test.dart [hunk 1]
```diff
@@ -309,6 +312,54 @@ void main() {
     expect(tester.takeException(), isNull);
   });
 
+  testWidgets('DataTable sort indicator orientation test', (WidgetTester tester) async {
```
Comment (darrenaustin) [inline packages/flutter/test/material/data_table_test.dart:315]: Minor nit:

```suggestion
  testWidgets('DataTable sort indicator orientation', (WidgetTester tester) async {
```

The 'test' here is redundant, as they are all tests 😄 .

**Review by darrenaustin — APPROVED** [review]
Nice job. Thanks for the contribution. I just had a minor nit on the name of the test, but otherwise it LGTM. Also you should update the description in the PR to include that you added a test to verify the arrow was pointing in the correct orientation.

**Comment by Wikiwix** [comment]
> Nice job. Thanks for the contribution. I just had a minor nit on the name of the test, but otherwise it LGTM. Also you should update the description in the PR to include that you added a test to verify the arrow was pointing in the correct orientation.

Thanks :)
Updated the description and commited your suggestion. (FYI: I actually felt the same as you about the test name, but I followed – what seemed to me, like – the *standard* of test names in the file 🙈 )

**Comment by darrenaustin** [comment]
> Updated the description and commited your suggestion. (FYI: I actually felt the same as you about the test name, but I followed – what seemed to me, like – the _standard_ of test names in the file 🙈 )

Ah, I didn't look at the context. Perhaps the original author had a different opinion, in which case you were probably right to follow what was there. At any rate, let's go with this for now. Once the tests pass I will get it merged. Thx.

**Comment by Wikiwix** [comment]
@darrenaustin Right now everything is green :)

**Pull request merged**